tomtom-navigation-core-examples/usecase/src/main/kotlin/com/example/usecase/MainViewModel.kt

```diff
@@ -0,0 +1,264 @@
+        mapLocationProvider.removeOnLocationUpdateListener(this)
+    }
+
+    fun registerLocationUpdatesListener() {
+        mapLocationProvider.addOnLocationUpdateListener(this)
+    }
+
+    /**
+     * Used to start navigation based on a tapped route, if navigation is not already running.
+     * - Then start the navigation using the selected route.
+     */
+    private val routeClickListener = RouteClickedListener { routingRoute, _ ->
+        if (!isNavigationRunning()) {
+            startNavigation(routingRoute)
+        }
+    }
+
+    private fun startNavigation(route: Route) {
+        setSimulationLocationProviderToNavigation(route)
+        setUpMapMatchedLocationProvider()
+        _requestNavigationStart.value = NavigationStartParameters(
+            routePlan = com.tomtom.sdk.navigation.RoutePlan(
+                route = route,
+                routePlanningOptions = checkNotNull(routePlanningOptions) {
+                    "RoutePlanningOptions is not initialized"
+                }),
+            tomTomNavigation = tomTomNavigation,
+            locationProvider = mapMatchedLocationProvider
+        )
+    }
+
+    /**
+     * Used to calculate a route using the following parameters:
+     * - InstructionPhoneticsType - This specifies whether to include phonetic transcriptions in the response.
+     * - ExtendedSections - This specifies whether to include extended guidance sections in the response, such as sections of type road shield, lane, and speed limit.
+     */
+    fun calculateRoute(origin: GeoPoint, destination: GeoPoint) {
+        val itinerary = Itinerary(origin = origin, destination = destination)
+        routePlanningOptions = RoutePlanningOptions(
+            itinerary = itinerary,
+            guidanceOptions = GuidanceOptions(
+                phoneticsType = InstructionPhoneticsType.Ipa,
+                extendedSections = ExtendedSections.All
+            ),
+            vehicle = Vehicle.Car()
+        )
+        routePlanner.planRoute(routePlanningOptions!!, routePlanningCallback)
+    }
+
+    /**
+     * The RoutePlanningCallback itself has three methods.
+     * - The `onFailure()` method is triggered if the request fails.
+     * - The `onSuccess()` method returns RoutePlanningResponse containing the routing results.
+     * - The `onRoutePlanned()` method is triggered when each route is successfully calculated.
+     *
+     * This example draws the first retrieved route on the map.
+     * You can show the overview of the added routes using the TomTomMap.zoomToRoutes(Int) method.
+     * Note that its padding parameter is expressed in pixels.
+     */
+    private val routePlanningCallback = object : RoutePlanningCallback {
+        override fun onSuccess(result: RoutePlanningResponse) {
+            val routePlan = RoutePlan(result.routes)
+            navigationVisualization.displayRoutePlan(routePlan)
+            navigationVisualization.selectRoute(routePlan.routes.first().id)
```

Comment:
You dont need to select the first route on the route plan. It will be selected by default.